.github/workflows/codeql.yml

```
@@ -0,0 +1,50 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+    - 'src/**'
+  pull_request:
+    branches: [ main ]
+    types: [opened, synchronize, reopened]
+    paths:
+    - 'src/**'
```

Adding this will allow the workflow to be triggered in PRs where the workflow itself is changed
```suggestion
    - 'src/**'
    - '‎.github/workflows/codeql.yml'
```